### DISCUSSION TITLE
How to avoid duplicate documentation?

### DISCUSSION THREAD

**Discussion by gnawin** [TulipaEnergy/TulipaEnergyModel.jl#484] [Ideas]
Referring to the PR #483, I think it would be good practice to reference docstring directly without creating duplicate content.

I would think using `@docs` is the way to go https://documenter.juliadocs.org/stable/man/syntax/#@docs-block.

But I got an error msg below:
```
┌ Error: duplicate docs found for 'TulipaEnergyModel.EnergyProblem' in src\reference.md:14-16
│ ```@autodocs
│ Modules = [TulipaEnergyModel]
│ ```
└ @ Documenter C:\Users\wangn\.julia\packages\Documenter\1HwWe\src\utilities\utilities.jl:44
[ Info: CrossReferences: building cross-references.
[ Info: CheckDocument: running document checks.
[ Info: Populate: populating indices.
ERROR: LoadError: `makedocs` encountered an error [:autodocs_block] -- terminating build before rendering.`
```
It suggests it conflicts with the use of `@autodocs` when we create the table of contents for the documentation. 

For this PR, it should be fine, but for any future docs it would be nice to make it work. I don't see a direct solution to this, or maybe I missed something. Any ideas?

**Comment by abelsiqueira** [comment]
I don't understand what was done. Where was it duplicated?

**Reply by gnawin** [reply]
In that PR, I copied what was in the docstring of `EnergyProblem` and manually pasted into the how-to section explaining the struct of `EnergyProblem`, so the same content appears twice. I would think the part in the how-to section should be referenced to the docstring.

**Reply by gnawin** [reply]
@abelsiqueira I added you as a reviewer so you may have a look 😃

**Reply by abelsiqueira** [reply]
Ah, got it. Checked online and apparently you can do this:

```@docs; canonical=false
TulipaEnergyModel.EnergyProblem
```

The `canonical=false` makes it so that this is considered the "extra" docs, and the `@autodocs` one is the official place. If it works, I think it is what we want.

**Reply by gnawin** [reply]
Works already!

But I was expecting to only get the text, not like the reference-style, with the box and names etc. The way it is now is inconsistent with the rest. 
![image](https://github.com/TulipaEnergy/TulipaEnergyModel.jl/assets/125902905/5c6a734a-5cc0-45ce-8b2c-65609380adcc)

**Reply by gnawin** [reply]
another pic showing how it looks with the rest:

![image](https://github.com/TulipaEnergy/TulipaEnergyModel.jl/assets/125902905/f0db9f2b-6bda-4da4-9612-3c8d4f0d8f5c)

**Reply by abelsiqueira** [reply]
Maybe something like

```@example
@doc EnergyProblem
```

might work? Would need to test some variation of this. `@doc` is part of Base, I think.

**Reply by gnawin** [reply]
Thank you. I've tried some workarounds (see below), but still not as satisfactory. For the time being, I will not block the PR #483. 

![image](https://github.com/TulipaEnergy/TulipaEnergyModel.jl/assets/125902905/6e5c5bbe-6a64-4ec8-a10d-c86f540534d8).

Would be happy if we continue exploring this topic.

**Comment by clizbe** [comment]
> Works already!
> 
> But I was expecting to only get the text, not like the reference-style, with the box and names etc. The way it is now is inconsistent with the rest. ![image](https://private-user-images.githubusercontent.com/125902905/306242795-5c6a734a-5cc0-45ce-8b2c-65609380adcc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1ODM0MjQsIm5iZiI6MTcxMjU4MzEyNCwicGF0aCI6Ii8xMjU5MDI5MDUvMzA2MjQyNzk1LTVjNmE3MzRhLTVjYzAtNDVjZS04YjJjLTY1NjA5MzgwYWRjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOFQxMzMyMDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZDcxMWI2ZjY3ZDk3MDNkMjdjYjM0NjBjOGY0MmViZGE2NTUyN2M0NjMxZDYzNDE2YzQwZDlkNzBhM2Y5NGEwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ZSt49PxGU9xc_Kj12IRPBR_sM9Uw9tBo3QYfjZmL1_k)

I prefer this one - maybe we can just remove the collapse arrow?

**Reply by gnawin** [reply]
yes, but how...?

**Comment by clizbe** [comment]
I asked Julia:
https://discourse.julialang.org/t/fix-nesting-or-disable-expansion-arrow-using-documenter-jl/117424

Otherwise I guess it should just be nested since it's better for maintenance.

**Reply by gnawin** [reply]
Thanks for posting!
